### PR TITLE
Slurm Array and SPM model

### PR DESCRIPTION
**Pull request by xutaima**

### CODE REVIEW

File: simuleval/options.py [hunk 1]
```diff
@@ -183,7 +189,7 @@ def general_parser():
 
 def add_slurm_args(parser):
     parser.add_argument(
-        "--slurm-partition", default="learnaccel,ust", help="Slurm partition."
+        "--slurm-partition", default="learnaccel,learnfair,seamless", help="Slurm partition."
```
Comment (annasun28) [inline simuleval/options.py:192]: since this is a general toolkit, maybe we shouldn't set defaults here?

File: simuleval/utils/slurm.py [hunk 2]
```diff
@@ -65,10 +80,33 @@ def submit_slurm_job(config_dict: Optional[Dict] = None) -> None:
             r"[^\"'\s]+\.py", f"{os.path.abspath(args.output)}/agent.py", command
         ).strip()
 
+    sweep_command = ""
+    sbatch_job_array_head = ""
```
Comment (annasun28) [inline simuleval/utils/slurm.py:84]: not blocking, but submitit has a nice interface for job arrays (https://github.com/facebookincubator/submitit/blob/main/docs/examples.md#job-arrays), if we want to migrate soon

**Review by annasun28 — APPROVED** [review]
thanks Xutai!

File: simuleval/utils/slurm.py [hunk 3]
```diff
@@ -77,18 +115,23 @@ def submit_slurm_job(config_dict: Optional[Dict] = None) -> None:
 #SBATCH --nodes=1
 #SBATCH --gpus-per-node=1
 #SBATCH --ntasks-per-node=8
-#SBATCH --output="{args.output}/slurm-%j.log"
+#SBATCH --output="{log_path}"
 #SBATCH --job-name="{args.slurm_job_name}"
+{sbatch_job_array_head}
 
+{job_array_configs}
```
Comment (annasun28) [inline simuleval/utils/slurm.py:122]: This is undefined when not using a job array

File: simuleval/options.py (hunk 1 above)
Comment (xutaima) [reply simuleval/options.py:192]: Maybe we can leave seamless because we are seamless team?

File: simuleval/utils/slurm.py (hunk 2 above)
Comment (xutaima) [reply simuleval/utils/slurm.py:84]: Thanks! Yeah will definitely visit this later. I admit that the current approach is pretty hacky ....

File: simuleval/options.py (hunk 1 above)
Comment (xutaima) [reply simuleval/options.py]: nvm, you are right. I remove the default partition.

**Review by annasun28 — APPROVED** [review]

**Comment by annasun28** [comment]
oh, looks like there's a test failure related to calling `set_target_spm_model` on `LogInstance`

**Pull request merged**